new-integration-tests/src/test/java/oracle/weblogic/kubernetes/assertions/impl/Kubernetes.java

```diff
@@ -146,6 +147,41 @@ public static boolean isPodTerminating(String namespace, String domainUid, Strin
     return status;
   }
 
+  /**
+   * Checks if a WebLogic server pod has been patched with an expected image.
+   *
+   * @param namespace Kubernetes namespace in which the pod is running
+   * @param domainUid label that the pod is decorated with
+   * @param podName name of the WebLogic server pod
+   * @param image name of the image to check for
+   * @return true if pod's image has been patched
+   * @throws ApiException when there is an error in querying the Kubernetes cluster
+   */
+  public static boolean podImagePatched(
+      String namespace,
+      String domainUid,
+      String podName,
+      String image
+  ) throws ApiException {
+    boolean podPatched = false;
+    String labelSelector = null;
+    if (domainUid != null) {
+      labelSelector = String.format("weblogic.domainUID in (%s)", domainUid);
+    }
+    V1Pod pod = getPod(namespace, labelSelector, podName);
+    if (pod != null && pod.getSpec() != null) {
+      List<V1Container> containers = pod.getSpec().getContainers();
```

Comment:
do we need a null check on containers?

Reply:
As far as I can tell, if there is no container, we'll get an empty list.